docker/docker_linux.go

```diff
@@ -52,6 +52,58 @@ func GetDockerStat() ([]CgroupDockerStat, error) {
 	return ret, nil
 }
 
+// Generates a mapping of PIDs to container metadata.
+func GetContainerStats() (map[string]ContainerStat, error) {
+	containerMap := make(map[string]ContainerStat)
+	path := common.HostSys("fs/cgroup/cpuacct")
+	if common.PathExists(path) {
+		contents, err := common.ReadDirectory(path)
+		if err != nil {
+			return nil, err
+		}
+
+		// If docker containers exist, collect their stats.
+		if common.StringsContains(contents, "docker") {
+			dockerStats, err := GetDockerStat()
+			if err != nil {
+				return nil, err
+			}
+
+			for _, dockerStat := range dockerStats {
+				if !dockerStat.Running {
+					continue
+				}
+
+				dockerPids, err := CgroupPIDDocker(dockerStat.ContainerID)
+				if err != nil {
+					continue
+				}
+
+				bytes, err = json.Marshal(dockerStat)
+				if err != nil {
+					continue
+				}
+
+				var jsonMap map[string]string
+				err = json.Unmarshal(bytes, &jsonMap)
```

Comment:
Doing a marshal then unmarshal like this is going to be pretty inefficient at higher container volumes (json is pretty slow in go). With an explicit struct you can avoid this altogether